### PR TITLE
Fix telemetry unit test timeouts

### LINKED ISSUE #9183: [UT Timeout] `getNpmVersion() is valid` -- `@react-native-windows/telemetry` -- `src/test/versionUtils.test.ts`
From CI, running on a quad-core machine with SSD, Node 16 with NPM 8.

https://dev.azure.com/ms/react-native-windows/_build/results?buildId=243798&view=logs&j=535198f9-5290-560b-cf5a-5b218edc34cc&t=d20bebb2-ce95-5535-0bb8-3ac674ec850d&l=2198

```
ERR!  FAIL  src/test/versionUtils.test.ts (7.5 s)
ERR!   √ getNodeVersion() is valid
ERR!   × getNpmVersion() is valid (5125 ms)
ERR!   √ getYarnVersion() is valid (1953 ms)
ERR!   √ getVisualStudioVersion() is valid (94 ms)
ERR!   √ getVersionOfNpmPackage() of empty string is null
ERR!   √ getVersionOfNpmPackage() of invalid package is null
ERR!   √ getVersionOfNpmPackage() of valid package is valid (16 ms)
ERR!   √ getVersionsOfNuGetPackages() of valid package in packages.config is valid (15 ms)
ERR!   √ getVersionsOfNuGetPackages() of invalid package in packages.config is invalid
ERR!   √ getVersionsOfNuGetPackages() of valid package in project is valid (16 ms)
ERR!   √ getVersionsOfNuGetPackages() of invalid package in project is invalid (16 ms)
ERR! 
ERR!   ● getNpmVersion() is valid
ERR! 
ERR!     : Timeout - Async callback was not invoked within the 5000 ms timeout specified by jest.setTimeout.Timeout - Async callback was not invoked within the 5000 ms timeout specified by jest.setTimeout.Error:
ERR! 
ERR!       32 | });
ERR!       33 |
ERR!     > 34 | test('getNpmVersion() is valid', async () => {
ERR!          | ^
ERR!       35 |   const node = await lookpath('npm');
ERR!       36 |   const version = await versionUtils.getNpmVersion();
ERR!       37 |   if (node) {
ERR! 
ERR!       at new Spec (../../../node_modules/jest-jasmine2/build/jasmine/Spec.js:116:22)
ERR!       at Object.<anonymous> (src/test/versionUtils.test.ts:34:1)

```

image

### PR DESCRIPTION
**Pull request by jonthysell**
Some tests were checking for the presence of npm/yarn before checking
its version. Since we almost always have those installed, I rearranged
the tests to look for the version first, and only if it returns null do
I check to confirm the tool is indeed missing.

Closes #9183

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/9193)

### CODE REVIEW

File: packages/@react-native-windows/telemetry/src/test/versionUtils.test.ts [hunk 1]
```diff
@@ -32,24 +32,22 @@ test('getNodeVersion() is valid', async () => {
...
 test('getNpmVersion() is valid', async () => {
-  const node = await lookpath('npm');
   const version = await versionUtils.getNpmVersion();
-  if (node) {
+  if (version) {
     expectValidVersion(version, true);
   } else {
-    console.log('npm not installed');
-    expect(version).toBeNull();
+    jest.setTimeout(10000); // Sometimes this lookup can run longer than the default 5000ms
```
Comment (NickGerleman) [inline packages/@react-native-windows/telemetry/src/test/versionUtils.test.ts:39]: This is icky. Why is it taking so long to locate npm in the first place?

**Review by NickGerleman — CHANGES_REQUESTED** [review]
This feels like fixing the test to ignore a defect in product code. We should have a better understanding of why it is taking over five seconds to lookup where npm lives, since that seems like a bug.

**Comment by jonthysell** [comment]
> This feels like fixing the test to ignore a defect in product code. We should have a better understanding of why it is taking over five seconds to lookup where npm lives, since that seems like a bug.

There is no defect in the product code. The test (and **only** the test) is using the `lookpath` library to check if the tool is missing, as confirmation that the version is null. Searching through all the paths on the agent takes a long time apparently.

I've changed the test to no longer confirm that tool is missing when the returned version is null.

File: packages/@react-native-windows/telemetry/src/test/versionUtils.test.ts (hunk 1 above)
Comment (jonthysell) [reply packages/@react-native-windows/telemetry/src/test/versionUtils.test.ts]: Removed.

**Review by NickGerleman — APPROVED** [review]

**Pull request merged**